torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -273,7 +273,29 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
       return shape;
     }
 
-    case aten::cat:
+    case aten::cat: {
+      // In JIT IR, aten::cat usually appears with the following nodes around
+      // it:
+      //   %dim : int = prim::Constant[value=0]()
+      //   %inputs : Tensor[] = prim::ListConstruct(%a, %b, ...)
+      //   %cat_output : Tensor = aten::cat(%inputs, %dim)
+      // Shapes of the input tensors could only differ at the dimension %dim.
+      // The sizes of the output tensor on that dimension is a sum of the
+      // corresponding sizes of the input tensors, the other dimension have the
+      // same sizes.
+      auto const& n = v->node();
+      auto inputs = n->inputs()[0]->node()->inputs();
```

Comment:
Does inputs()[0] necessarily have to have inputs?  What happens if it doesn't?

Reply:
I thought that it would fire an assert, but after re-examination I found that it would just cause an out of bound access. I will change `[0]` to `.at(0)`, which has the boundary check. Thanks for noticing this!